src/App.tsx

```diff
@@ -1,8 +1,83 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { TodoList } from './TodoList';
 import './App.css';
+import {
+  getUsers, getTodos, ApdateTodo, Users, Todos,
+} from './api';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const App: React.FC = () => {
+  const [todos, setTodos] = useState<ApdateTodo[]>([]);
+  const [isLoaded, setLoaded] = useState(false);
+  const [isLoading, setLoading] = useState(false);
+
+  const loadAllTodos = () => {
+    let updateTotos: ApdateTodo[] = [];
+
+    Promise.all([getTodos(), getUsers()])
+      .then(data => {
+        updateTotos = data[0].map((todo: Todos) => ({
+          ...todo, user: data[1].find((user: Users) => user.id === todo.userId),
+        }));
+      }).then(() => {
+        setTodos(updateTotos);
+        setLoading(false);
+        setLoaded(true);
+      });
+  };
+
+  const handleLoading = () => {
+
+
+    setLoading(true);
+      setTimeout(() => {
+        loadAllTodos();
+      }, 1000);
+
+  };
+
+  const sortByTodosName = () => {
+    const sortedTodos = [...todos].sort((todoPrew, todoCurr) => (
```

Comment:
```suggestion
    const sortedTodos = [...todos].sort((a, b) => (
```
для сорту прийнято так писати